File: KadirliApp/KadirliApp/Domain/Entities/Pharmacy.swift
import Foundation

struct Pharmacy: Identifiable, Codable {
    let id: UUID
    let name: String
    let phone: String?
    let address: String?
    let region: String
    let latitude: Double?
    let longitude: Double?
    let dutyDate: String // YYYY-MM-DD formatında String
    
    enum CodingKeys: String, CodingKey {
        case id, name, phone, address, region, latitude, longitude
        case dutyDate = "duty_date"
    }
}

File: KadirliApp/API_Deneme/Models/Pharmacy.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API_Deneme.Models
{
    public class Pharmacy
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(20)]
        public string? Phone { get; set; }
        
        [MaxLength(500)]
        public string? Address { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Region { get; set; } = string.Empty;
        
        public double? Latitude { get; set; }
        
        public double? Longitude { get; set; }
        
        [Required]
        [Column("duty_date")]
        [MaxLength(10)]
        public string DutyDate { get; set; } = string.Empty; // YYYY-MM-DD formatında
    }
}

